cogs/resourcecog.py

```diff
@@ -0,0 +1,169 @@
+                print("Remember to !setreviewchannel")
+
+    @commands.command()
+    @commands.has_role("Bot Commander")
+    async def setresourcechannel(self, ctx, channel_id):
+        """Enter ID of channel to set as resource channel"""
+        rsc_channel = await self.bot.fetch_channel(channel_id)
+
+        if rsc_channel is None:
+            await ctx.send("Invalid channel id")
+            return
+
+        with open("config.json", "r") as f:
+            cfg = json.load(f)
+            cfg["resourceChannel"] = channel_id
+
+        with open("config.json", "w") as f:
+            json.dump(cfg, f)
+
+        self.resource_channel_id = channel_id
+        await ctx.send(f"{rsc_channel.name} has been set as resources channel")
+
+    @commands.command()
+    @commands.has_role("Bot Commander")
+    async def setreviewchannel(self, ctx, channel_id):
+        """Enter ID of channel to set as review channel"""
+        rvw_channel = await self.bot.fetch_channel(channel_id)
+
+        if rvw_channel is None:
+            await ctx.send("Invalid channel id")
+            return
+
+        with open("config.json", "r") as f:
+            cfg = json.load(f)
+            cfg["reviewChannel"] = channel_id
+
+        with open("config.json", "w") as f:
+            json.dump(cfg, f)
+
+        self.review_channel_id = channel_id
+        await ctx.send(f"{rvw_channel.name} has been set as resources channel")
+
+    @commands.command()
+    async def suggestresource(self, ctx, rscname, rsclink, rscdesc, *, rsctags):
+        """Suggest a resource with the name, link, description (in quotes), and tags"""
+        rvw_channel = await self.bot.fetch_channel(self.review_channel_id)
+        message = await rvw_channel.send(
+            embed=create_embed(
+                ctx.message.author.name, rscname, rsclink, rscdesc, rsctags
+            )
+        )
+        await message.add_reaction("✔️")
+        await message.add_reaction("❎")
+        cfg = jsonopen()
+
+        cfg[str(message.id)] = {}
+        rinfo = cfg[str(message.id)]
+        rinfo["name"] = rscname
+        rinfo["link"] = rsclink
+        rinfo["desc"] = rscdesc
+        rinfo["tags"] = rsctags
+        rinfo["msgauth"] = str(ctx.message.author.name)
+
+        jsonsave(cfg)
+
+    @commands.command()
+    @commands.has_role("Bot Commander")
+    async def rscedit(self, ctx, msg_id, editedmsg):
+        """Edit a resource description by inputting the id of the message and the new description"""
+        cfg = jsonopen()
+        try:
+            cfg[str(msg_id)]["desc"] = editedmsg
+        except KeyError:
+            await ctx.send("That is not a valid ID")
+        channel = self.bot.get_channel(int(self.review_channel_id))
+        message = await channel.fetch_message(int(msg_id))
+        await message.edit(embed=edit_embed(cfg[str(msg_id)]))
+        jsonsave(cfg)
+
+    @commands.Cog.listener()
+    async def on_raw_reaction_add(self, payload):
```

Comment:
Did you use the raw reaction add event listener because you saw it being used elsewhere? You don't actually need to use this listener instead of the nicer reaction add listener as long as you don't need to listen for reactions to messages that were sent _before_ the bot came online.